jupyter/analysis-automation/BasicExample.ipynb

```diff
@@ -0,0 +1,456 @@
+    "data_api = tdmreader.DataApi()\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "import scrapbook as sb"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_property(element, property_name):\n",
+    "    \"\"\"Gets a property of the given element.\n",
+    "    \n",
+    "    The element can be a file, channel group, or channel.\n",
+    "    \n",
+    "    Args:\n",
+    "        element: Element to get the property from.\n",
+    "        property_name: Name of the property to get.\n",
+    "        \n",
+    "    Returns:\n",
+    "        The according property of the element or ``None`` if the property doesn't exist.\n",
+    "    \"\"\"\n",
+    "    return next((e.value for e in element.properties.properties if e.name == property_name), None)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Retrieve Metadata with a Data Link\n",
+    "\n",
+    "A data link is the input for each __Analysis Automation procedure__ that uses a query to collect specific data items. A `data_link` contains a list of one or more elements that point to a list of files, channel groups, or channels (depending on the query result type).\n",
+    "\n",
+    "This example shows how the Metadata API accesses the `file_info` structure from the file, through the `groups`, and down to the `channels` level.\n",
+    "\n",
+    "This example calculates the absolute minimum and absolute maximum value of all channels in each group and displays these values in the report."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_links = ni_analysis_automation[\"data_links\"]\n",
+    "file_ids = [d[\"fileId\"] for d in data_links]\n",
+    "file_infos = await metadata_api.get_multiple_file_info(tdmreader.FileList(file_ids))\n",
+    "file_info = file_infos[0]\n",
+    "\n",
+    "test_file_name = get_property(file_info, \"name\")\n",
+    "program_name = get_property(file_info, \"Test~Procedure\")\n",
+    "\n",
+    "group_names = []\n",
+    "channels = []\n",
+    "formatted_properties = []\n",
+    "\n",
+    "for group in file_info.groups:\n",
+    "    group_names.append(group.name)\n",
+    "    channels.append(group.channels)\n",
+    "    max_values_of_group = []    \n",
+    "    min_values_of_group = []    \n",
+    "    mean_values_of_group = []    \n",
+    "\n",
+    "    for channel in group.channels:\n",
+    "        minimum = float(get_property(channel, \"minimum\") or \"NaN\")\n",
+    "        maximum = float(get_property(channel, \"maximum\") or \"NaN\")\n",
+    "        mean_values_of_group.append((minimum + maximum) / 2)\n",
+    "        max_values_of_group.append(maximum)\n",
+    "        min_values_of_group.append(minimum)\n",
+    "\n",
+    "        \n",
+    "    # Calculate statistical values from metadata\n",
```

Comment:
I like this code comment.  I would like to see more of these for each block of code within the cell.

Reply:
Thanks for the suggestion. We might add some more comments with a later update.